components/NavLayout/SideBar.js

```diff
@@ -1,12 +1,183 @@
+    }),
+  }),
+);
 
 export default function SideBar({ children, ...props }) {
+  const theme = useTheme();
+  const [open, setOpen] = useState(false);
+
+  const handleDrawerOpen = () => {
+    setOpen(true);
+  };
+
+  const handleDrawerClose = () => {
+    setOpen(false);
+  };
+
+  // hard coded data, eventually change sampleUser to [username]
+  const pageUrls = {
+    0: "/sampleUser/my-classes",
+    1: "/sampleUser/calendar",
+    2: "/sampleUser/notifications",
+    3: "/sampleUser/class-catalog"
+  }
+
+  // TODO: change "Current User" to [username]'s name
+  // TODO: change avatar src to dynamically render user's profile picture
   return (
-    <nav className={styles.nav}>
-      <h3>My Classes</h3>
-      <h3>Calendar</h3>
-      <h3>Notifications</h3>
-      <h3>Class Catalog</h3>
-    </nav>
+    <Box sx={{ display: 'flex' }}>
+      <CssBaseline />
+      <AppBar position="fixed" open={open} className={styles.topbar}>
+        <Toolbar>
+          <IconButton
+            color="inherit"
+            aria-label="open drawer"
+            onClick={handleDrawerOpen}
+            edge="start"
+            sx={{
+              marginRight: 5,
+              ...(open && { display: 'none' }),
+            }}
+          >
+            <MenuIcon />
+          </IconButton>
+          <Typography className={styles.siteTitle} variant="h5" noWrap component="div">
+            ClassMates
+          </Typography>
+          <Container className={styles.profileIcon} sx={{ flexGrow: 0 }}>
+            <Typography className={styles.profileName}>Current User</Typography>
+            <Avatar
+              className={styles.profileAvatar}
+              alt="Remy Sharp"
+              src="https://www.vhv.rs/dpng/d/426-4264903_user-avatar-png-picture-avatar-profile-dummy-transparent.png"
+              sx={{ width: 50, height: 50 }}
+            />
+          </Container>
+        </Toolbar>
+      </AppBar>
+      <Drawer className='drawer' variant="permanent" open={open} >
+        <DrawerHeader>
+          <IconButton onClick={handleDrawerClose}>
+            {theme.direction === 'rtl' ? <ChevronRightIcon style={{ color: '#FFFFFF' }} /> : <ChevronLeftIcon style={{ color: '#FFFFFF' }} />}
+          </IconButton>
+        </DrawerHeader>
+
+        <List>
+          {['My Classes', 'Calendar', 'Notifications', 'Class Catalog'].map((text, index) => (
+            <ListItemButton
+              key={text}
+              sx={{
+                minHeight: 48,
+                justifyContent: open ? 'initial' : 'center',
+                px: 2.5,
+              }}
+            >
+              <ListItemIcon
+                sx={{
+                  minWidth: 0,
+                  mr: open ? 3 : 'auto',
+                  justifyContent: 'center',
+                }}
+              >
+                {index === 0 && <Link href={pageUrls[index]}><ClassOutlinedIcon style={{ color: '#FFFFFF' }} /></Link>}
```

Comment:
Maybe instead of having 4 checks inside the map, we can make an object like you did for the urls with the components as values. This way instead of doing 4 checks each map iteration, it'll just be an object lookup. Could even edit the existing object above to incorporate the icon components
EX)
`
const pageUrls = {
    0: { url:"/sampleUser/my-classes", icon: <ClassOutlinedIcon style{{...}} /> }, ...
  }
`
Might not be the best solution but look into cleaning up.